docs/source/essentials/normalized-cache.mdx

```diff
@@ -0,0 +1,350 @@
+
+// Build the ApolloClient
+val apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(cacheFactory)
+  .build())
+```
+
+```java:title=Java
+// Create a 10MB NormalizedCacheFactory
+NormalizedCacheFactory cacheFactory = new LruNormalizedCacheFactory(EvictionPolicy.builder().maxSizeBytes(10 * 1024 * 1024).build());
+
+// Build the ApolloClient
+ApolloClient apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(cacheFactory)
+  .build();
+```
+
+</MultiCodeBlock>
+
+## Persisting your data in a SQLite database
+
+If the amount of data you store becomes too big to fit in memory or if you want your data to persist between app restarts, you can also use a `SqlNormalizedCacheFactory`. A `SqlNormalizedCacheFactory` will store your data in a [SQLDelight](https://github.com/cashapp/sqldelight) database and is defined in a separate dependency:
+
+```kotlin:title=build.gradle.kts
+dependencies {
+  implementation("com.apollographql.apollo:apollo-normalized-cache-sqlite:x.y.z")
+}
+```
+
+**Note:** The `apollo-normalized-cache-sqlite` dependency has Kotlin multiplatform support and has multiple variants (`-jvm`, `-android`, `-ios-arm64`,...). If you are targetting Android and using custom `buildTypes`, you will need to help Gradle resolve the correct artifact by defining [matchingFallbacks](https://developer.android.com/studio/build/dependencies?authuser=2#resolve_matching_errors):
+
+<MultiCodeBlock>
+
+```kotlin:title=build.gradle
+android {
+  buildTypes {
+    create("custom") {
+      // your code...
+      matchingFallbacks = listOf("debug")
+    }
+  }
+}
+```
+
+```groovy:title=build.gradle
+android {
+  buildTypes {
+    custom {
+      // your code...
+      matchingFallbacks = ["debug"]
+    }
+  }
+}
+```
+
+</MultiCodeBlock>
+
+Once the dependency is added, create the `SqlNormalizedCacheFactory`:
+
+<MultiCodeBlock>
+
+```kotlin
+// Android
+val sqlNormalizedCacheFactory = SqlNormalizedCacheFactory(context, "apollo.db")
+// JVM
+val sqlNormalizedCacheFactory = SqlNormalizedCacheFactory("jdbc:sqlite:apollo.db")
+// iOS
+val sqlNormalizedCacheFactory = SqlNormalizedCacheFactory("apollo.db")
+
+// Build the ApolloClient
+val apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(sqlNormalizedCacheFactory)
+  .build())
+```
+
+```java
+// Android
+SqlNormalizedCacheFactory sqlNormalizedCacheFactory = new SqlNormalizedCacheFactory(context, "apollo.db")
+// JVM
+SqlNormalizedCacheFactory sqlNormalizedCacheFactory = new SqlNormalizedCacheFactory("jdbc:sqlite:apollo.db")
+
+// Build the ApolloClient
+ApolloClient apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(sqlNormalizedCacheFactory)
+  .build();
+```
+
+</MultiCodeBlock>
+
+## Chaining caches
+
+To get the best of both caches, you can chain a `LruNormalizedCacheFactory` with a `SqlNormalizedCacheFactory`:
```

Comment:
a -> an before `Lru`

Reply:
Today [I learnt](https://english.stackexchange.com/questions/378279/why-use-an-before-a-word-that-starts-with-a-l) I should use `an` before `l` if `el` is pronounced literally ("el")